src/graph.jl

```diff
@@ -0,0 +1,349 @@
+using DataFrames
+using GraphDataFrameBridge
+using MetaGraphs
+using CSV
+using LightGraphs
+using GraphPlot
+using Random
+using DataFramesMeta
+
+function load_graph_data(input_data)
+    
+    return input_data
+end
+    
+"""    generate_synthetic_vertices(vertex_type_defs::String, output_path::String)
+
+Generate synthetic test data. The synthetic vertices are returned as a dataframe 
+that can be used for testing/debugging/developing the knowledge graph.
+
+see also: generate_synthetic_edges
+"""
+function generate_synthetic_vertices(vertex_type_defs::String, output_path::String)
+
+    
+    vertex_types = CSV.read(vertex_type_defs)
+    
+    synth_vertices_df = []
+    
+    for vertex in eachrow(vertex_types)
+        
+        # Generate a synthetic name; hash the concatenation of name||type to get a unique vertex id
+        v_name = string("synth_", vertex.v_type, "_", randstring(MersenneTwister(), 'a':'z', 6)) 
+        v_hash = hash(string(v_name, vertex.v_type))       
+        synth_v_attrs = NamedTuple{(:v_hash, :v_name, :v_type)}(("$v_hash", v_name, vertex.v_type))
+        
+        push!(synth_vertices_df, synth_v_attrs)
+        
+    end
+    
+    open(output_path, "w") do io
+        print(io, repr(load_graph_data(synth_vertices_df)))
+    end
+
+    #CSV.write(output_path, synth_vertices_df)
+    @info "Synthetic vertex dataframe generated and saved as a Julia file."
+    
+    return DataFrame(synth_vertices_df)
+
+end
+
+
+function generate_synthetic_edges(edge_type_defs::String, synth_vertex_df::DataFrame, output_path::String)
+    
+    """
+    This function ingests a synthetic vertex data frame and generates synthetic edges that are consistent with edge src/dst vertex type constraints. 
+    This synthetic vertex dataframe can be used for testing/debugging/developing the knowledge graph.
+    """
+    
+    edge_types = CSV.read(edge_type_defs)
+
+    synth_edges_df = []
+    
+    for edge_row in eachrow(edge_types)
+
+        src_row = @linq synth_vertex_df |> 
```

Comment:
yay`@linq`!

I'm not familiar enough with the different query packages, but the most famous is https://github.com/queryverse/Query.jl, is there a reason to use DataFrameMeta instead?

Reply:
No particular reason; I think I came across @linq first and it seemed pretty straightforward. I can swap out these functions for Query functions if we want.

Reply:
if it works it works and we can stick with it until there is motivation to change. Lets stick with `@ling` for this PR.

btw, you should put macros in backticks, to save the poor souls `@show, @test, @linq, @info, @warn, ...` from getting random julia related notifications.